File: 29230-Lista_5-AA_basico-2016.2/P3-Binary_search.py
def binarySearch(element, elements = []):
    left = 0
    right = len(elements) - 1
    middle = (right + left) / 2

    while left <= right:
        if (elements[middle] == element):
            return middle
        else:
            if (elements[middle] < element):
                left = middle + 1
                middle = (right + left) / 2
            else:
                right = middle - 1
                middle = (right + left) / 2
    return -1


n, q = map(int, raw_input().split())

elements = map(int, raw_input().split())

for query in xrange(q):
    print binarySearch(int(raw_input()), elements)

File: 29230-Lista_5-AA_basico-2016.2/P3-Binary_search.cpp
#include<stdio.h>

int binarySearch(long int *array, long int element, long int right) {
    int left = 0;
    int backup = -1;

    while (left <= right) {
        int middle = left + (right - left) / 2;

        if(array[middle] == element) {
            backup = middle;
            right = middle -1;
        } else {
            if (array[middle] < element) {
                left = middle + 1;
            }else {
                right = middle - 1;
            }
        }
    }
    return backup;
}

int main() {
    long int n, q, result;

    scanf("%ld %ld", &n, &q);
    long int array [100000], results[500000];


    int count = 0;
    for (int i = 0; i < n; i++){
        long long int number;
        scanf("%lli", &number);
        if (number < 0){
            count++;
        }
        array[i] = number;
    }


    long long int element;
    for (int i = 0; i < q; i++) {
        scanf ("%li", &results[i]);
    }
    for (int i = 0; i < q; i ++) {
        result = binarySearch(array, results[i], n);
        printf("%ld\n", result);
    }
    return 0;
}






















